### PR TITLE
Fix "No updates to be performed." throwing Error code 1

### LINKED ISSUE #2981: No updates are to be performed - exit code 1
# This is a Feature Proposal

## Description

```
Serverless: Updating Stack...
 
  Serverless Error ---------------------------------------
 
     No updates are to be performed.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
 
  Your Environment Information -----------------------------
     OS:                 linux
     Node Version:       7.2.1
     Serverless Version: 1.4.0
 
error Command failed with exit code 1.
```

I'm not sure how the rest of us feel about this, but I did not expect the `exit code 1`. Do we want to change this? In our case it failed our CI deployment.

### PR DESCRIPTION
**Pull request by paulSambolin**
## What did you implement:

Closes #2981

## How did you implement it:
Catch the error, then check for specific error message "No updates to be performed".  continue execution if it is this message, otherwise throw the error to be caught by the next Error handler

## How can we verify it:
Re-deploy an already deployed service without any changes and verify that it does not error out with Error code 1

```yml
service: service

provider:
  name: aws
  runtime: nodejs4.3

# remove need for functions so that a re-deploy will bring up the error
#functions:
#  hello:
#    handler: handler.hello

resources:
  Resources:
    S3Bucket:
      Type: "AWS::S3::Bucket"
```

## Todos:

- [ ] Write tests
- [ ] Write documentation
- [x] Fix linting errors
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config/commands/resources
- [x] Enable ["Allow edits from maintainers"](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) for this PR
- [ ] Change ready for review message below

***Is this ready for review?:*** NO
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/plugins/aws/lib/updateStack.js [hunk 1]
```diff
@@ -81,7 +81,14 @@ module.exports = {
       params,
       this.options.stage,
       this.options.region)
-      .then((cfData) => this.monitorStack('update', cfData));
+      .then((cfData) => this.monitorStack('update', cfData))
+      .catch((e) => {
+        console.log(e);
```
Comment (pmuens) [inline lib/plugins/aws/lib/updateStack.js]: This should be removed

File: lib/plugins/aws/lib/updateStack.js [hunk 2]
```diff
@@ -81,7 +81,14 @@ module.exports = {
...
+        if (e.message === 'No updates are to be performed.') {
```
Comment (mthenw) [inline lib/plugins/aws/lib/updateStack.js]: Can we define that as a const?

**Comment by mthenw** [comment]
Created separate PR with fixes.

**Comment by mthenw** [comment]
https://github.com/serverless/serverless/pull/3047

**Pull request closed without merging**